src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -785,7 +785,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Get
                 If _lazyAllSyntaxTrees.IsDefault Then
                     Dim builder = ArrayBuilder(Of SyntaxTree).GetInstance()
-                    builder.AddRange(_rootNamespaces.Keys)
+                    builder.AddRange(_syntaxTrees)
```

Comment:
The spec for `Compilation.SyntaxTrees` is that it is the sequence of trees the compilation was created with. I don't know why VB's property has a different specification that appears to allow nondeterminism. It is in fact deterministic, as it must be to have reliable semantics. `_syntaxTrees` is in the order provided to the compilation. `_rootNamespaces` on the other hand is a `Dictionary` and so its keys change order from compilation to compilation.

There are even assertions that the order is deterministic, such as this one in the constructor:

```vb
            Debug.Assert(syntaxTrees.All(Function(tree) syntaxTrees(syntaxTreeOrdinalMap(tree)) Is tree))
```

I'll remove the part of the spec that allows nondeterminism.